sdk/tests/utilities/test_id_generator.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+from . import IdGenerator
+from parameterized import parameterized
+
+
+class IdGeneratorTests(unittest.TestCase):
+
+    def test_use_default_scope_when_not_specified(self):
+        id_generator = IdGenerator()
+        self.assertEqual("sdk_example", id_generator.default_scope)
+
+    @parameterized.expand([
+        ["custom-scope"],
+        [""]
+    ])
+    def test_uses_provided_scope(self, scope):
+        id_generator = IdGenerator(scope)
+        self.assertEqual(scope, id_generator.default_scope)
+
+    def test_explicit_none_scope_uses_default(self):
+        id_generator = IdGenerator(None)
+        self.assertEqual("sdk_example", id_generator.default_scope)
+
+    @parameterized.expand([
+        ["default scope", None, "sdk_example"],
```

Comment:
Is it worth setting a class var `default_scope = "sdk_example"` for use across this test class?

Reply:
done